lib/galaxy/model/__init__.py

```diff
@@ -2457,6 +2458,46 @@ def __init__(self, user, group):
         self.group = group
 
 
+class Notification(Base, Dictifiable, RepresentById):
+    __tablename__ = "notification"
+
+    id = Column(Integer, primary_key=True)
+    create_time = Column(DateTime, default=now)
+    update_time = Column(DateTime, default=now, onupdate=now)
+    publication_time = Column(DateTime, default=now)
+    expiration_time = Column(DateTime, default=now() + timedelta(days=30 * 6))
+    source = Column(String(32), index=True)
+    category = Column(String(64), index=True)
+    variant = Column(String(16), index=True)
```

Comment:
Does the `variant` field imply a scale? ("Defines the 'importance' of the notification.") If so, maybe use an integer instead? Not only will that give us more flexibility (avoid the need for "high" vs. "somewhat high" vs. "very very high", etc.), we also get natural ordering.

Reply:
The idea was more along the line of something like "info", "warning", "urgent", etc. that can be used to render or highlight the notifications differently (colors, icons, etc.). I think the ordering should be always by date, but this `variant` can be used for filtering, i.e. display only "urgent" notifications.
Of course, these enums can be mapped as integers in the DB if this is more desirable, I just thought they are more readable as strings, but I'm happy to change that if this is more optimum :)

Reply:
Ah, I see. Those values are more informative than a simple integer scale - let's keep them as strings then.
With regard to the database, it's all the same here. Thanks for the explanation!